tests/test_web_sendfile_functional.py

```diff
@@ -266,35 +285,34 @@ async def handler(request):
 
     resp = await client.get("/")
     assert resp.status == 200
-    body = await resp.read()
-    assert body == b"hello aiohttp\n"
-    assert resp.headers["Content-Type"] == "text/plain"
     assert resp.headers.get("Content-Encoding") == "gzip"
+    assert resp.headers["Content-Type"] == "text/plain"
+    assert await resp.read() == HELLO_AIOHTTP
     resp.close()
     await resp.release()
     await client.close()
 
 
+@pytest.mark.parametrize(
+    ("hello_txt", "expect_encoding"), [["gzip"] * 2], indirect=["hello_txt"]
+)
 async def test_static_file_with_content_encoding(
-    aiohttp_client: Any, sender: Any
+    hello_txt: pathlib.Path, aiohttp_client: Any, sender: Any, expect_encoding: str
 ) -> None:
-    filepath = pathlib.Path(__file__).parent / "hello.txt.gz"
+    """Test requesting of static compressed files returns correct content type and encoding."""
 
     async def handler(request):
-        return sender(filepath)
+        return sender(hello_txt)
 
     app = web.Application()
     app.router.add_get("/", handler)
     client = await aiohttp_client(app)
 
     resp = await client.get("/")
-    assert 200 == resp.status
-    body = await resp.read()
-    assert b"hello aiohttp\n" == body
-    ct = resp.headers["CONTENT-TYPE"]
-    assert "text/plain" == ct
-    encoding = resp.headers["CONTENT-ENCODING"]
-    assert "gzip" == encoding
+    assert resp.status == 200
+    assert resp.headers.get("Content-Encoding") == expect_encoding
+    assert resp.headers["Content-Type"] == "text/plain"
+    assert await resp.read() == HELLO_AIOHTTP
```

Comment:
If you don't mind, I'd like to keep the comparisons inverted:
```suggestion
    assert 200 == resp.status
    assert expect_encoding == resp.headers.get("Content-Encoding")
    assert "text/plain" == resp.headers["Content-Type"]
    assert HELLO_AIOHTTP == await resp.read()
```
This helps avoid awkward keyword combos like `assert await`.

@Dreamsorcerer @bdraco any objections to having this style? I wasn't the one to introduce it but it seems like a good idea to follow it.

Reply:
I find it slightly awkward, as I'm rather accustomed to the expected value being on the right. I don't really care much though.